src/helpers/common.ts

```diff
@@ -151,6 +151,25 @@ export function callbackifyGenerator<T>(generatorFn: () => AsyncGenerator<T>) {
   };
 }
 
+export function callbackifyStream<T>(
+  stream: Readable,
+  callbackFn: Callback<T>,
+) {
+  stream.on('data', (data) => callbackFn(null, data));
+  stream.on('error', (err) => (callbackFn as ErrorCallback)(err));
+  stream.on('finish', () => (callbackFn as DataCallback<T | null>)(null, null));
+}
```

Comment:
Based on naming, it should be a factory function. (look at built-in NodeJS util `.callbackify` -> returns a function`

```typescript
export function callbackifyStream<T>(callbackFn: Callback<T>) {
  return (stream: Readable) => {
    stream.on('data', (data) => callbackFn(null, data));
    stream.on('error', (err) => (callbackFn as ErrorCallback)(err));
    stream.on('finish', () => (callbackFn as DataCallback<T | null>)(null, null));
  }
}
```

Reply:
Will do 👍 , note your example here has the arguments reversed.